tripal_chado_views/tripal_chado_views.views.inc

```diff
@@ -596,88 +596,130 @@ function tripal_chado_views_views_data_tripal_views_tables($data) {
-              'label' => t("@base_title => Node", array('@base_title' => $base_title)),
-              'real field' => 'nid',
-              'base' => 'node',
-              'base field' => 'nid'
+  // ADD IN ENTITIES JOINS & RELATIONSHIPS
+  // D7 @todo: Create custom handler to allow join from Entity => Base (ie: organism)
+  //           with the addition of a single relationship
+  // D7 @todo: Create custom handler to allow join from Base (ie: organism)
+  //           with the addition of a single relationship
+  // D7 @todo: Add support for Mview <-> Entity joins and relationships
+  $bundle_query = db_select('chado_bundle', 'CB');
+  $bundle_query->fields('CB', array('bundle_id', 'data_table'));
+  $bundle_query->join('tripal_bundle', 'TB', 'CB.bundle_id = TB.id');
+  $bundle_query->fields('TB', array('name', 'label'));
+  $bundle_query->join('tripal_term', 'TT', 'TB.term_id = TT.id');
+  $bundle_query->fields('TT', array('accession'));
+  $bundle_query->join('tripal_vocab', 'TV', 'TT.vocab_id = TV.id');
+  $bundle_query->fields('TV', array('vocabulary'));
+  $bundles = $bundle_query->execute();
+  // Look for the record ID in the appropriate chado table.
+  foreach ($bundles as $bundle) {
+    // Ids we'll use for queries.
+    // Ex.: 'stock'.
+    $base_table = $bundle->data_table;
+    // Ex.: 'stock_id'.
+    $base_id_field = $base_table . '_id';
+    // Ex.:'Chado Stock'.
+    $base_table_label = 'Chado ' . ucwords(str_replace('_', ' ', $base_table));
+    // Ex.: 'chado_bio_data_21'.
+    $linker_table = 'chado_' . $bundle->name;
+    // Ex.: 'Germplasm Accession'
+    $bundle_label = $bundle->label;
+    // Ex.: 'CO_010__0000044'
+    $bundle_internal_name = $bundle->vocabulary . '__' . $bundle->accession;
+
+    // Add in joins to the tripal_entity tables if the Chado schema is local.
+    $is_local = isset($GLOBALS["chado_is_local"]) && $GLOBALS["chado_is_local"];
+    if ($is_local) {
+      if (db_table_exists($linker_table)) {
+        // Adds in a chado base table => entity relationship.
+        // This allows controlled joining to multiple entities per line.
+        // Use Case:  link to feature and organism entities on a feature listing.
+        // D7 todo: a custom relationship handler to get from feature.organism_id => organism node
+        //      without 1st needing to add relationship to organism table.
+        $data[$linker_table]['table']['join'][$base_table] = array(
+          'left_field' => $base_id_field,
+          'field' => 'record_id',
+        );
+        $data[$linker_table]['table']['join']['tripal_entity'] = array(
+          'left_field' => 'id',
+          'field' => 'entity_id',
+        );
+        $data[$linker_table][$base_id_field] = array(
+          'group' => $base_table_label,
+          'title' => $base_table_label,
+          'help' => t(
+            "Links @base_table_label to its @bundle_label.",
```

Comment:
I think we should mention Tripal Content Type (Entity) to make it easier to search for all Tripal Content type relationships.
```suggestion
            "Links @base_table_label to its Tripal Content Type '@bundle_label'.",
```